Expo-video tvOS title, artist, and subTitle

HADeveloper opened the pull request:
# Why

This PR is to add the ability to customize the video player metadata (especially on tvos).

# How

I added the metadata types and followed the docs here: https://developer.apple.com/documentation/avkit/customizing-the-tvos-playback-experience

# Test Plan

Since this is for TVOS, to test this, you play a video using expo-video and set the metadata parameters and you should see them show up when the player is playing the video, especially when paused.
![Screenshot 2024-12-12 at 11 44 17 AM](https://github.com/user-attachments/assets/ae0764e7-20d2-4d46-bc66-cb4e5770960a)

# Checklist

- [*] I added a `changelog.md` entry and rebuilt the package sources according to [this short guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [*] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).
- [*] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

Commit:
Added subtitle to  VideoPlayer.types.ts

Commit:
Added subtitle to  VideoPlayer.types.ts

Commit:
Added metadata configuration to the vide player in VideoPlayer.swift

Commit:
Added metadata configuration to the vide player in VideoPlayer.swift

Commit:
Update CHANGELOG.md

Commit:
Update CHANGELOG.md

Commit:
built

Commit:
built

behenate reviewed (changes requested):
Hi before I proceed with an actual review, it seems like your fork is incorrectly rebased from main (there is a lot of additional changes from other PRs), also please fix all of the SwiftLint violations instead of marking them as resolved, the code has to be compliant with our SwiftLint rules 😄

douglowder commented:
> Hi before I proceed with an actual review, it seems like your fork is incorrectly rebased from main (there is a lot of additional changes from other PRs), also please fix all of the SwiftLint violations instead of marking them as resolved, the code has to be compliant with our SwiftLint rules 😄

I concur with @behenate 's comment above. When checking SwiftLint, you should use [this config file](https://github.com/expo/expo/blob/main/.swiftlint.yml) at the top of our monorepo. You can reach out to me on the Expo Discord `building-for-tv` channel if you have any issues with the rebase or with SwiftLint.

Commit:
Reigned in the modifications, linted, and rebased

Commit:
Reigned in the modifications, linted, and rebased

Commit:
rebased and linted

Commit:
Merge branch 'expo:main' into expo-video-upgrade

Commit:
fixed linting using correct linting rules

Commit:
fixed linting using correct linting rules

Commit:
Merge branch 'expo-video-upgrade' of https://github.com/HADeveloper/expo into expo-video-upgrade

Commit:
Merge branch 'expo:main' into expo-video-upgrade

HADeveloper commented:
I think I have it working now.
Please check @behenate @douglowder 

Thanks!
(I'm still learning the process and any guidance is greatly appreciated)

douglowder committed:
Add subTitle type to VideoPlayer.types.ts

douglowder committed:
Add subTitle type to VideoPlayer.types.ts

douglowder committed:
Add metadata configuration to Swift implementation

douglowder committed:
Add metadata configuration to Swift implementation

douglowder committed:
Build JS

douglowder committed:
Build JS

douglowder committed:
CHANGELOG

douglowder committed:
CHANGELOG

douglowder commented:
> I think I have it working now. Please check @behenate @douglowder
> 
> Thanks! (I'm still learning the process and any guidance is greatly appreciated)

@HADeveloper there are merge conflicts and SwiftLint errors still in the PR.

To unblock you, I created a new branch from main, cherry-picked your changes, and resolved conflicts and fixed SwiftLint issues (including two that I found that actually were not part of your PR 😄 )

Please take a look at https://github.com/expo/expo/tree/doug/pr-33639 -- if you are happy with those changes, please sync your PR branch to that, and merge issues should be resolved. @behenate

jonsamp committed:
[docs] Adds update job parameters (#33790)

Kudo committed:
[config-types] sync 52.0.0 schema (#33786)

# Why

following up #33637 and now to sync the schema

# How

`yarn generate && yarn lint --fix && yarn prepare` 
manually remove the synced `packagerOpts` which is deprecated already

alanjhughes committed:
[splashscreen] Remove setting `appearance` on default icon (#33799)

alanjhughes committed:
[ios][splashscreen] Show splashscreen on reload (#33793)

marklawlor committed:
[go] Bump screens to 4.4.0 and navigation to 7.0.14 (#33758)

# Why

Bumps `react-native-screens` and `react-navigation/*` to latest verions

---------

Co-authored-by: Kudo Chien <kudo@expo.dev>

Kudo committed:
[fingerprint] fix expo-modules-autolinking path in pnpm isolated mode (#33818)

# Why

try to fix `npx expo-modules-autolinking` resolving issue on pnpm isolated mode

# How

we should resolve the path through `expo -> expo-modules-autolinking`. this pr tries to use the same logic originally to resolve the `expo-modules-autolinking` version to achieve that.
- rename 'ExpoVersions.ts` to `ExpoResolver.ts` and add two more `resolveExpoAutolinkingPackageRoot()` and `resolveExpoAutolinkingCliPath()` helper. it will cache the resolved result to prevent duplicated work.
- replace `npx expo-modules-autolinking` to `node resolveExpoAutolinkingCliPath(projectRoot)`

Kudo committed:
[fingerprint] fix ExpoConfigLoader broken from invalid JSON stdout (#33821)

# Why

fixes #33798

# How

introduce a `spawnWithIpcAsync` that uses ipc channel than stdout to get result from ExpoConfigLoader

Kudo committed:
[fingerprint] fix absolute paths in aarProjects autolinking (#33826)

# Why

fixes #33725

# How

use relative paths for `aarProjects` autolinking data

Kudo committed:
[fingerprint] fix console.log cli accident change (#33828)

# Why

revert my accident console.log change from #33821

# How

revert the console.log change

Kudo committed:
et ssbc -b sdk-52 [skip ci]

amandeepmittal committed:
[docs] Add info on how to enable Array syntax usage in Shared routes (#30964)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Fixes the issue on how to enable **Array syntax** in an app when using
shared routes.

Refs
[Slack](https://exponent-internal.slack.com/archives/C1QMWQ1P0/p1723151618462689)

# How

<!--
How did you build this feature or fix this bug and why?
-->

Changes:
- On Shared routes page, add info on how to enable array syntax by using
`initialRouteName` in the dynamic layout route file. It continues using
the example defined in the Array syntax section. This will act as a
source of truth on how to enable this setting.
- On Router Settings page, remove the info that explains the context
about using `initialRouteName` for array syntax. This is because the
**Array syntax** section can act as a source of truth in itself and we
should provide the info on how to enable it within that section.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Run docs locally and see:
http://localhost:3002/router/advanced/shared-routes/#arrays.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

scottwoodall committed:
Update authentication.mdx (#33831)

Minor grammar fix.

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

# How

<!--
How did you build this feature or fix this bug and why?
-->

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

tsapeta committed:
[contacts] Expose ContactAccessButton from SwiftUI (#33782)

dependabot[bot] committed:
Bump rexml from 3.2.9 to 3.3.9 (#32416)

Bumps [rexml](https://github.com/ruby/rexml) from 3.2.9 to 3.3.9.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/ruby/rexml/releases">rexml's
releases</a>.</em></p>
<blockquote>
<h2>REXML 3.3.9 - 2024-10-24</h2>
<h3>Improvements</h3>
<ul>
<li>Improved performance.
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/210">GH-210</a></li>
<li>Patch by NAITOH Jun.</li>
</ul>
</li>
</ul>
<h3>Fixes</h3>
<ul>
<li>
<p>Fixed a parse bug for text only invalid XML.</p>
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/215">GH-215</a></li>
<li>Patch by NAITOH Jun.</li>
</ul>
</li>
<li>
<p>Fixed a parse bug that <code>&amp;#0x...;</code> is accepted as a
character
reference.</p>
</li>
</ul>
<h3>Thanks</h3>
<ul>
<li>NAITOH Jun</li>
</ul>
<h2>REXML 3.3.8 - 2024-09-29</h2>
<h3>Improvements</h3>
<ul>
<li>SAX2: Improve parse performance.
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/207">GH-207</a></li>
<li>Patch by NAITOH Jun.</li>
</ul>
</li>
</ul>
<h3>Fixes</h3>
<ul>
<li>Fixed a bug that unexpected attribute namespace conflict error for
the predefined &quot;xml&quot; namespace is reported.
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/208">GH-208</a></li>
<li>Patch by KITAITI Makoto</li>
</ul>
</li>
</ul>
<h3>Thanks</h3>
<ul>
<li>
<p>NAITOH Jun</p>
</li>
<li>
<p>KITAITI Makoto</p>
</li>
</ul>
<h2>REXML 3.3.7 - 2024-09-04</h2>
<h3>Improvements</h3>
<ul>
<li>Added local entity expansion limit methods
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/192">GH-192</a></li>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/202">GH-202</a></li>
<li>Reported by takuya kodama.</li>
</ul>
</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/ruby/rexml/blob/master/NEWS.md">rexml's
changelog</a>.</em></p>
<blockquote>
<h2>3.3.9 - 2024-10-24 {#version-3-3-9}</h2>
<h3>Improvements</h3>
<ul>
<li>Improved performance.
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/210">GH-210</a></li>
<li>Patch by NAITOH Jun.</li>
</ul>
</li>
</ul>
<h3>Fixes</h3>
<ul>
<li>
<p>Fixed a parse bug for text only invalid XML.</p>
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/215">GH-215</a></li>
<li>Patch by NAITOH Jun.</li>
</ul>
</li>
<li>
<p>Fixed a parse bug that <code>&amp;#0x...;</code> is accepted as a
character
reference.</p>
</li>
</ul>
<h3>Thanks</h3>
<ul>
<li>NAITOH Jun</li>
</ul>
<h2>3.3.8 - 2024-09-29 {#version-3-3-8}</h2>
<h3>Improvements</h3>
<ul>
<li>SAX2: Improve parse performance.
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/207">GH-207</a></li>
<li>Patch by NAITOH Jun.</li>
</ul>
</li>
</ul>
<h3>Fixes</h3>
<ul>
<li>Fixed a bug that unexpected attribute namespace conflict error for
the predefined &quot;xml&quot; namespace is reported.
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/208">GH-208</a></li>
<li>Patch by KITAITI Makoto</li>
</ul>
</li>
</ul>
<h3>Thanks</h3>
<ul>
<li>
<p>NAITOH Jun</p>
</li>
<li>
<p>KITAITI Makoto</p>
</li>
</ul>
<h2>3.3.7 - 2024-09-04 {#version-3-3-7}</h2>
<h3>Improvements</h3>
<ul>
<li>Added local entity expansion limit methods
<ul>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/192">GH-192</a></li>
<li><a
href="https://redirect.github.com/ruby/rexml/issues/202">GH-202</a></li>
<li>Reported by takuya kodama.</li>
</ul>
</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/ruby/rexml/commit/38eaa86ac7abe0d31cf49d8df57ad239fdeb80e9"><code>38eaa86</code></a>
Add 3.3.9 entry</li>
<li><a
href="https://github.com/ruby/rexml/commit/ce59f2eb1aeb371fe1643414f06618dbe031979f"><code>ce59f2e</code></a>
parser: fix a bug that &amp;#0x...; is accepted as a character
reference</li>
<li><a
href="https://github.com/ruby/rexml/commit/a09646d395a07399cbf9bc3bc8d6d8bb1d13ecea"><code>a09646d</code></a>
test: fix indent</li>
<li><a
href="https://github.com/ruby/rexml/commit/cf0fb9c9ca3dc0d725c8e4644aa0e728025f42ce"><code>cf0fb9c</code></a>
Fix <code>IOSource#readline</code> for <code>@pending_buffer</code> (<a
href="https://redirect.github.com/ruby/rexml/issues/215">#215</a>)</li>
<li><a
href="https://github.com/ruby/rexml/commit/1d0c362526f6e25e2abcd13e2fcefcc718c20e78"><code>1d0c362</code></a>
Optimize <code>IOSource#read_until</code> method (<a
href="https://redirect.github.com/ruby/rexml/issues/210">#210</a>)</li>
<li><a
href="https://github.com/ruby/rexml/commit/622011f25ac1519fd553d6c56da52d7eba14a787"><code>622011f</code></a>
Bump version</li>
<li><a
href="https://github.com/ruby/rexml/commit/036d50851ce091c797db0b9ba3ed8e5a39c3918c"><code>036d508</code></a>
test: avoid using needless non ASCII characters</li>
<li><a
href="https://github.com/ruby/rexml/commit/4197054a19e65511fb51983518a134a5c65aa840"><code>4197054</code></a>
Add 3.3.8 entry</li>
<li><a
href="https://github.com/ruby/rexml/commit/78f8712dccad773a51dc5eef31c02d523e994570"><code>78f8712</code></a>
Fix handling with &quot;xml:&quot; prefixed namespace (<a
href="https://redirect.github.com/ruby/rexml/issues/208">#208</a>)</li>
<li><a
href="https://github.com/ruby/rexml/commit/2e1cd64f2f9c0667a840a0e31f9bb99f9e1c2b33"><code>2e1cd64</code></a>
Optimize SAX2Parser#get_namespace (<a
href="https://redirect.github.com/ruby/rexml/issues/207">#207</a>)</li>
<li>Additional commits viewable in <a
href="https://github.com/ruby/rexml/compare/v3.2.9...v3.3.9">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=rexml&package-manager=bundler&previous-version=3.2.9&new-version=3.3.9)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)
You can disable automated security fix PRs for this repo from the
[Security Alerts page](https://github.com/expo/expo/network/alerts).

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Simek committed:
[docs] APISectionClasses: display methods using nested presentation (#33820)

dependabot[bot] committed:
[go] Bump path-to-regexp from 1.8.0 to 1.9.0 (#31422)

Bumps [path-to-regexp](https://github.com/pillarjs/path-to-regexp) from
1.8.0 to 1.9.0.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/pillarjs/path-to-regexp/releases">path-to-regexp's
releases</a>.</em></p>
<blockquote>
<h2>Fix backtracking in 1.x</h2>
<p><strong>Fixed</strong></p>
<ul>
<li>Add backtrack protection to 1.x release (<a
href="https://redirect.github.com/pillarjs/path-to-regexp/issues/320">#320</a>)
925ac8e</li>
<li>Fix
<code>re.exec(&amp;[#39](https://github.com/pillarjs/path-to-regexp/issues/39);/test/route&amp;[#39](https://github.com/pillarjs/path-to-regexp/issues/39);)</code>
result (<a
href="https://redirect.github.com/pillarjs/path-to-regexp/issues/267">#267</a>)
32a14b0</li>
</ul>
<p><a
href="https://github.com/pillarjs/path-to-regexp/compare/v1.8.0...v1.9.0">https://github.com/pillarjs/path-to-regexp/compare/v1.8.0...v1.9.0</a></p>
</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/pillarjs/path-to-regexp/commit/c75eb105b2a177822c1dfd58e0e032320cd868ff"><code>c75eb10</code></a>
1.9.0</li>
<li><a
href="https://github.com/pillarjs/path-to-regexp/commit/925ac8e3c5780b02f58cbd4e52f95da8ad2ac485"><code>925ac8e</code></a>
Add backtrack protection to 1.x release (<a
href="https://redirect.github.com/pillarjs/path-to-regexp/issues/320">#320</a>)</li>
<li><a
href="https://github.com/pillarjs/path-to-regexp/commit/32a14b0185ab23c1afae6dd7b624a01ce68fd470"><code>32a14b0</code></a>
Fix <code>re.exec('/test/route')</code> result (<a
href="https://redirect.github.com/pillarjs/path-to-regexp/issues/267">#267</a>)</li>
<li>See full diff in <a
href="https://github.com/pillarjs/path-to-regexp/compare/v1.8.0...v1.9.0">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=path-to-regexp&package-manager=npm_and_yarn&previous-version=1.8.0&new-version=1.9.0)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

You can trigger a rebase of this PR by commenting `@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)
You can disable automated security fix PRs for this repo from the
[Security Alerts page](https://github.com/expo/expo/network/alerts).

</details>

> **Note**
> Automatic rebases have been disabled on this pull request as it has
been open for over 30 days.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Simek commented on packages/expo-video/CHANGELOG.md:
```diff
@@ -15,10 +16,7 @@
 
 ### 💡 Others
 
-- Fixed `generateThumbnailsAsync` not being available on Android in the types. ([#33491](https://github.com/expo/expo/pull/33491) by [@hirbod](https://github.com/hirbod))
-- Run VideoManager.setAppropriateAudioSessionOrWarn on a different queue for a lower load on the main thread. ([#33127](https://github.com/expo/expo/pull/33127) by [@behenate](https://github.com/behenate))
-
-## 2.0.2 - 2024-11-29
```
Looks like rebase artifact, probably should be restored.

Simek commented on packages/expo-video/src/VideoPlayer.types.ts:
```diff
@@ -272,6 +251,11 @@ export type VideoSource =
        * When undefined the player will display information contained in the video metadata.
        * @platform android
        * @platform ios
+<<<<<<< HEAD
+       * @platform tvos
+=======
+       * @default tvos
+>>>>>>> main
```
Unresolved merge conflict.

Simek commented on packages/expo-video/src/VideoPlayer.types.ts:
```diff
@@ -315,6 +301,17 @@ export type VideoMetadata = {
    */
   artwork?: string;
+  /**
+<<<<<<< HEAD
+   * The text dispayed above the title on the player
+   * @platform android
+   * @platform ios
+=======
+   * The subTitle of the video.
+>>>>>>> main
```
Unresolved merge conflict.

rtorrente committed:
[sqlite] fix kv-store promises not well transfered when using AsyncSt… (#33847)

# Why

Related to investigations and bug found in
https://github.com/expo/expo/issues/33754.
The promises of the upper function are not transfered to the aliased
function when using AsyncStorage compatible api alias. Await call on
this function are useless and this can trigger database locked when
multiple setItem are trigerred.

# How

Add await and return on the aliased function when needed

Kudo committed:
[sqlite] fix kv-store database lock error (#33834)

# Why

fixes #33754

# How

if people mix async/sync call, the global `db` variable could be
corrupted. rather than having a mutex that supports both async and sync,
i decide to use the sync `getDbSync()` simply. it might be slower from
async operations, but won't be too slow though.

amandeepmittal committed:
[docs] Whitelist global tutorial code annotation class in eslint config (#33848)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

![CleanShot 2024-12-26 at 19 19
32](https://github.com/user-attachments/assets/67deebdb-a333-4cf5-979a-3d909f372de8)


# How

<!--
How did you build this feature or fix this bug and why?
-->

Whitelist `tutorial-code-annotation` class under
`tailwindcss/no-custom-classname` in `.eslintrc.cjs` file.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Run `yarn run lint`, and there will be no warnings anymore.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

amandeepmittal committed:
[docs] Remove pleasantries in multiple docs (#33846)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Remove unnecessary pleasantries from multiple docs since the
documentation is provided to inform developers.

# How

<!--
How did you build this feature or fix this bug and why?
-->

- By going through multiple docs and removing words like "please" and
"please note". Replace the context with appropriate verbiage (where
necessary).
- Remove an older comment about opening a docs issue in this GitHub
repository from app config schema page.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Docs have been tested locally, and all checks are being passed.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

Simek committed:
[face-detector] remove deprecated package and usages (#32471)

alanjhughes committed:
[android][audio] Accept null source in audio player (#33854)

Kudo committed:
[dev-menu] fix perf monitor on ios (#33855)

# Why

fix performance monitor ui dismissed right immediately after toggling on.
close ENG-14589

# How

as @alanjhughes pointed out, the root cause is from
https://github.com/facebook/react-native/commit/16775215d5a734288da7147db348824395198407.
this pr tries to hide and resign the DevMenuWindow key status
temporarily during show the performance monitor.

---------

Co-authored-by: Alan Hughes <30924086+alanjhughes@users.noreply.github.com>

EvanBacon committed:
chore(expo-router): Add less aggressive babel plugin migration warning. (#33640)

# Why

- Make it a bit easier for users upgrading from older versions of Expo
Router (v2 to v4+)

---------

Co-authored-by: Expo Bot <34669131+expo-bot@users.noreply.github.com>

rtorrente committed:
[sqlite] Fixed `database is locked` error while using `kv-store multiGet (#33873)

# Why

As seen in this issue https://github.com/expo/expo/issues/33754, we have
some strange low-level crash behavior when using this function. It
appears that the use of an exclusive transaction blocking the database
to all other read and write requests is not necessarily the most
appropriate on a read-only request.

# How

Remove withExclusiveTransactionAsync on multiGet function

amandeepmittal committed:
[docs] Fix minor issues (#33883)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

- Following our writing style guide, change folder to directory.
- Fix typos and minor verbiage improvements.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

amandeepmittal committed:
[docs] Create a custom APIBox component to apply padding styles for manual usage (#33880)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Fix ENG-14593

# How

<!--
How did you build this feature or fix this bug and why?
-->

- Create a custom wrapper component around `APIBox` so we can expand
this component with more styles, if required, for manual usage.
- Update all `APIBox` component manual usage with the new component.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

![CleanShot 2024-12-30 at 02 37
02](https://github.com/user-attachments/assets/c69f1e3d-952f-45e4-b948-08ae3d4bdac2)


# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

amandeepmittal committed:
[docs] Update Box component styles (#33881)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Currently, the `Box` component used in the OAuth provider reference has
broken styles.

 
![CleanShot 2024-12-30 at 02 40
56](https://github.com/user-attachments/assets/1cb053ed-3565-4870-9694-600d4edde888)


# How

<!--
How did you build this feature or fix this bug and why?
-->

Add the padding to match the updated styles from `APIBox` after its
recent update and also update the `Table` styles since margin bottom
wasn't applied previously.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

![CleanShot 2024-12-30 at 02 59
24@2x](https://github.com/user-attachments/assets/b0291776-d292-4e8b-a945-3abfd379af00)


# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

rtorrente committed:
[sqlite] Enforce input validations in `kv-store` operations (#33874)

# Why

Following the debug in issue https://github.com/expo/expo/issues/33754
(but not directly linked)
I don't think we should rely solely on static typing to validate inputs.
I had a problem with a library that specified string output and produced
an object under certain conditions, and debugging was complicated.
Like the AsyncStorage library, I think it's better to throw a determined
error if the input is bad than to leave strange behavior at the native
level.

# How

Add input validation on every function

amandeepmittal committed:
[docs] Upgrade Vale dev dependency, update comment delimiter usage and vale config (#33888)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

With Vale `3.9.2`, using comment delimiter to switch the vale rule in a
`.mdx` file now supports MDX comment syntax. More info about syntax
usage and comment delimiters in Vale documentation:
https://vale.sh/docs/keys/commentdelimiters.

# How

<!--
How did you build this feature or fix this bug and why?
-->

- Upgrade Vale dev dependency to use `3.9.2`
- Update vale configuration to add `CommentDelimiters` for mdx file type
- Update comment delimiter usage in multiple mdx files.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

After applying these changes, run `yarn run lint-prose`, and there will
be no warnings or errors from the Vale lint tool.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

leopic committed:
[fingerprint] correct typo in fingerprint cli (#33887)

tsapeta committed:
[core][iOS] Support dispatching events by SwiftUI views (#33860)

amandeepmittal committed:
[docs] Update babel config references (#33903)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

- Update multiple docs to add a link to the Babel config reference page.
- In the React compiler guide, add instructions to create a
`babel.config.js` file since the file isn't included in SDK 52 and above
default projects.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

All links and updated info has been verified by running docs locally.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

amandeepmittal committed:
[docs] Use complete option name instead of shorthand (#33904)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Use complete option name instead of shorthand when describing the
`--clear` option from Expo CLI. This is our recommended way of
describing a command in docs.

# How

<!--
How did you build this feature or fix this bug and why?
-->

Update both instances of where shorthand `-c` option was described.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

amandeepmittal committed:
[docs] Apply APIBox formatting to package.json reference (#33900)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Based on feedback from @Simek.

# How

<!--
How did you build this feature or fix this bug and why?
-->

Use `PaddedAPIBox` component.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

## Preview

![CleanShot 2024-12-31 at 22 07
30@2x](https://github.com/user-attachments/assets/caf5d6fb-1e13-41c7-befc-7e1695a6104b)


# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

Simek committed:
[docs] switch to MDX v3, update setup related dependencies (#33898)

Simek committed:
[docs] APISection: display single line / non-block examples inline (#33899)

amandeepmittal committed:
[router][docs] Fix broken React Navigation docs links in Expo Router reference (#33902)

<!-- disable:changelog-checks -->

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

- Fix broken React Navigation docs links in the Expo Router reference
- Found there was a typo in the `rel` property example
- Update navigation prop to called navigation object


# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Verify updated links are working.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

alanjhughes committed:
[expo-go] Fix reloading when typing in text input (#33917)

amandeepmittal committed:
Update link title

Commit:
Resolved merge conflicts and synced with doug/pr-33639

Commit:
Merge branch 'expo-video-upgrade' of https://github.com/HADeveloper/expo into expo-video-upgrade

Commit:
Update VideoPlayer.swift to remove artifacts

Commit:
Update VideoPlayer.swift to remove artifacts

Commit:
Update CHANGELOG.md added back in

Commit:
Update CHANGELOG.md added back in

HADeveloper commented:
I think I have everything right now. @douglowder

behenate commented:
@HADeveloper 
Please notice that in @douglowder branch there is only [11 changed files with 128 additions and 11 deletions](https://github.com/expo/expo/compare/main...doug/pr-33639#:~:text=with%20128%20additions%20and%2011%20deletions.)
Whereas if you look in the top right of this PR there is over 17 thousand lines changed in 38 files
<img width="801" alt="image" src="https://github.com/user-attachments/assets/0c5cbafc-4ee4-4ba0-9302-b7d877feef48" />
This means that most likely at one point something went wrong when merging the main branch into your branch (you should prefer rebasing over merging, because when merging stuff like this sometimes happens). You will likely solve those issues by fetching the newest version of `expo:main` and rebasing your branch onto it.

If you can't get it to work you can always try creating a fresh branch from the current `expo:main` version, pasting the appropriate changes and opening a new PR 😄 (I'd lie if I said I never did that when things got FUBAR'd 😅 )

behenate committed:
[video][ios] Fix `AVPlayer` not deallocating when the player is unmounted regression (#33922)

# Why

Fixes a regression accidentaly introduced in this PR
https://github.com/expo/expo/pull/33123
Fixes https://github.com/expo/expo/issues/33804

# How

When replacing with `nil` use a strong capture to the `pointer` to be
sure that the `AVPlayer` receives the message to replace the current
item even when `replaceCurrentItem` is called from the `deinit` block.

# Test Plan

Tested in BareExpo in iOS simulator

tsapeta committed:
[core][iOS] Add `UnimplementedExpoView` as a stub for SwiftUI on the old architecture (#33901)

Commit:
[docs] Highlighted a new import in the getting started tutorial (#33923)

The "useRef" hook was added in this section however it wasn't
highlighted.

Highlighted it and added the correct tooltip to align it with the rest
of the doc.

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->
Folks who are new to the world to react/react native like me use the
highlights to follow along and understand what changes are being made in
the tutorial.

Explaning the import would help others realise the change they are
making and hopefully understand it a bit better.
# How

<!--
How did you build this feature or fix this bug and why?
-->
I came across this while I was following the tutorial to build my first
expo app!

Fixed the issue by adding the a tooltip and highlighting the import
(similar to the rest of the docs)
# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Here's a screenshot of the proposed fix: 


![image](https://github.com/user-attachments/assets/c14e86c9-5526-4bdb-b951-403ef259bbd2)

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

---------

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

amandeepmittal committed:
[docs] Fix internal links and redirects (#33920)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Fix internal links and redirects based on Google Search Console not
found report. While navigating the list, I've found that some
server-side redirects were missing for the removed pages and some
redirects were outdated, so to fix them, I updated them to a currently
appropriate URL from the docs.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Running locally and testing some of the updated/added URLs. All tests
are passing.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

amandeepmittal committed:
[docs] Fix APIBox usage for app config schema (#33921)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Follow-up #33880

![CleanShot 2025-01-02 at 19 07
38@2x](https://github.com/user-attachments/assets/48aded6e-7751-4f74-8a97-e49d7161ee25)


# How

<!--
How did you build this feature or fix this bug and why?
-->

Add the padding to the `APIBox` component used inside
`AppConfigSchemaTable` component.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Run `yarn run test` to make sure all tests are passing locally.

## Preview

![CleanShot 2025-01-02 at 19 09
04@2x](https://github.com/user-attachments/assets/2e2ca62c-4a5b-4763-b759-3de42276f7b6)

![CleanShot 2025-01-02 at 19 08
54@2x](https://github.com/user-attachments/assets/4a538724-7d59-4718-84fa-b4b07f1fbdbd)


# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

pjdemers committed:
[packages][ios]Fix ScreenOrientation.addOrientationChangeListener() Freezes iOS Devices  (#33867)

entiendoNull committed:
[ios][expo-media-library] Introducing pairedVideoAsset (#33274)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Currently, when using `expo-media-library`, we don't have access to the
video asset of Live Photos, which would be pretty sweet to have now that
`expo-live-photo` was recently released. This PR attempts to bring this
functionality to the Media Library as well.

I requested this feature and @tsapeta suggested I could try open a PR
for it. Regardless how this goes, I had a lot of fun with this 😄

# How
<!--
How did you build this feature or fix this bug and why?
-->

The implementation adds support for Live Photos by:

- Detecting Live Photos using PHAsset's mediaSubtypes
- Exposing the pairedVideo through a new `pairedVideoAsset`-property in
the `AssetInfo`

I found it reasonable to expose the `pairedVideoAsset` through
`getAssetInfoAsync` - this new property is added to the response of it.
Currently, this property is automatically populated if the image has a
media subtype of `livePhoto`. Would it be better to make this opt-in
instead?

The `PairedVideoAsset` type is designed to look like the
`ImagePickerAsset` type for familiarity.


https://github.com/user-attachments/assets/2812076b-ff7e-49bf-9e41-c960d6657b14

I have not committed the changes to the screen in
`native-components-list` (which you see in the video). That was just a
test I did.

It should also be said that this is my first PR ever touching Swift. I
am not very familiar with it, and not sure about best practices or the
conventions that you may have - so please review it thorough! 😅 Also,
any feedback that you might have is much appreciated - I'd really love
to learn more.

If it's worth continuing with this PR I should also
- update docs
- add a screen to `native-components-list` describing the functionality.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

I tested this through Bare-Expo on my phone and it seems to work well. 
However, many of the tests for the Media Library seems to fail when I
run them. I will have to look into this.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

---------

Co-authored-by: Tomasz Sapeta <tomasz.sapeta@me.com>

rush2di committed:
[docs] small typo fix in custom tab layouts guide (#33925)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

# How

<!--
How did you build this feature or fix this bug and why?
-->

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

alanjhughes committed:
[audio] Fix issues with `replace` method (#33708)

EvanBacon committed:
fix(cli): Recurse server action exports to collect all references. (#33934)

# Why

- Some server actions can import others, these will need a recursive
bundling process to collect.

---------

Co-authored-by: Expo Bot <34669131+expo-bot@users.noreply.github.com>

amandeepmittal committed:
[docs] Fix typos in multiple docs (#33928)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

While going through multiple doc, found a couple of typos which are
fixed in this PR.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

N/A

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

EvanBacon committed:
fix(babel): fix server action extraction position (#33933)

# Why

The server actions were being extracted to a single location after the
imports which could lead to nested server actions not being initiated at
the right time. This PR addresses this issue by moving the extraction to
be before the nearest path in the global scope. This is something I've
patched around in the past when using libraries like `ai`.

Fix ENG-14602

# Test Plan

- Added test cases for the syntax.
- Added an extra pass to the tests to ensure the output JS is valid.

---------

Co-authored-by: Expo Bot <34669131+expo-bot@users.noreply.github.com>

amandeepmittal committed:
[docs] Update podcasts and live streams sections in additional resources (#33924)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

New podcasts appearances and live streams from the Expo team in last one
month.

# How

<!--
How did you build this feature or fix this bug and why?
-->

Update podcasts and live streams sections in additional resources.


# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

![CleanShot 2025-01-03 at 17 32
26](https://github.com/user-attachments/assets/3507aec6-6ca7-4947-b364-0dc7afb7e74d)

![CleanShot 2025-01-03 at 21 30
59](https://github.com/user-attachments/assets/ddfdda72-26df-4bda-b7d1-9a92fc58c3be)


# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

---------

Co-authored-by: Bartosz Kaszubowski <gosimek@gmail.com>

alanjhughes committed:
[audio] Minor refactor and docs (#33940)

LeonDvlpmnt committed:
[docs] change fingerprintExperimental reference to fingerprint (#33959)

# Why

`fingerprintExperimental` was changed to `fingerprint` in the SDK 51
release (https://github.com/expo/expo/pull/28220)

# How

<!--
How did you build this feature or fix this bug and why?
-->

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

amandeepmittal committed:
[docs] Update app config section title for consistency in API reference (#33958)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Currently, there are two different section titles in API docs used:
![CleanShot 2025-01-06 at 15 50
02@2x](https://github.com/user-attachments/assets/b5413eb4-d409-404c-845b-54af3cdaa0a7)

![CleanShot 2025-01-06 at 15 48
42@2x](https://github.com/user-attachments/assets/a529d0e6-82ce-44ff-8439-9e199215a604)

Since we have started to use "Configuration in app config" because "app
config" itself represents different file extensions, this PR updates the
left-over section titles to the updated verbiage for consistency.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Run docs locally and visit an API doc:
http://localhost:3001/versions/latest/sdk/video/

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [ ] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

tharakadesilva committed:
[bun] Add support for text-based lockfile format (#33825)

# Why

Bun v1.1.39 introduced a new text-based lockfile format (bun.lock) as an
alternative to the binary format (bun.lockb). This change adds support
for detecting and using both formats.

https://bun.sh/blog/bun-lock-text-lockfile

Similar to: https://github.com/expo/eas-cli/issues/2782 (I am not sure
how EAS resolves the package manager, hopefully through Expo?)

# How

The changes were implemented by:

1. Adding support for both lockfile formats in the package manager
resolution logic:
   - Added `BUN_TEXT_LOCK_FILE` constant for the new 'bun.lock' format
   - Updated `resolvePackageManager` to check for both lockfile formats
- Modified the lockFiles record to support multiple lockfile paths per
package manager

2. Updated documentation:
   - Added new section about text-based lockfile in using-bun.mdx
   - Updated CLI help text to reflect support for both formats
   - Updated EAS build documentation to mention both lockfile formats

3. Added comprehensive test coverage:
   - Tests for text lockfile only
   - Tests for binary lockfile only
   - Tests for both lockfiles present
   - Tests for monorepo workspace scenarios

4. Updated cleanup utilities to handle both lockfile formats:
- Modified `reinstallPackagesAsync` to clean up both `bun.lock` and
`bun.lockb`
   - Updated gitignore to include both lockfile formats

# Test Plan

1. Basic functionality test:

```bash
expo/expo/packages/@expo/cli/build/bin/cli install @supabase/supabase-js
```

- Without changes: Incorrectly picks up npm as the package manager
- With changes: Correctly identifies bun as the package manager when
either `bun.lock` or `bun.lockb` is present

2. Automated tests:
- Added comprehensive test suite in `nodeManagers-test.ts` covering:
  - Package manager resolution with text lockfile
  - Package manager resolution with binary lockfile
  - Package manager resolution with both lockfiles
  - Monorepo workspace scenarios
  - Various lockfile combinations

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [x] I added a `changelog.md` entry and rebuilt the package sources
according to [this short
guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

---------

Co-authored-by: Cedric van Putten <github@cedric.dev>

Commit:
Merge branch 'expo-video-upgrade' of https://github.com/HADeveloper/expo into expo-video-upgrade

HADeveloper commented:
I think it is getting worse
